### PR TITLE
Add execution history for worker

### PR DESCRIPTION
**Pull request by takonomura**
Add pages to show execution history for worker. You can check executed jobs at specific time.

#### Worker Logs page

worker_logs_page

#### Worker Logs Timeline page

worker_timeline_page

### CODE REVIEW

**Comment by ganmacs** [comment]
@cookpad/dev-infra Review please.

File: app/controllers/kuroko2/worker_logs_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    set_period
+    @logs = @logs.where('created_at < ?', @end_at).where('finished_at > ? OR finished_at IS NULL', @start_at)
+  end
+
+  private
+
+  def set_period
+    @end_at = begin params[:end_at].to_datetime rescue Time.current end
+
+    @start_at = begin params[:start_at].to_datetime rescue
```
Comment (ganmacs) [inline app/controllers/kuroko2/worker_logs_controller.rb]: (ditto) please specify error type

File: db/migrate/031_create_worker_logs.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class CreateWorkerLogs < ActiveRecord::Migration[5.0]
+  def change
+    create_table :worker_logs do |t|
+      t.string :hostname, limit: 180, null: false
```
Comment (ganmacs) [inline db/migrate/031_create_worker_logs.rb]: Doesn't it need to add index to hostname column?

File: spec/controllers/worker_logs_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+  describe '#dataset' do
+    subject! { get :dataset, xhr: true }
+
+    it do
+      expect(response).to have_http_status(:ok)
+
+      expect(assigns(:logs)).to match_array logs
+      expect(assigns(:end_at)).not_to be_nil
+      expect(assigns(:start_at)).to eq assigns(:end_at) - 1.hour
```
Comment (ganmacs) [inline spec/controllers/worker_logs_controller_spec.rb]: [nits] 
I think following code is more unambiguous than this code.

```ruby
expect(assigns(:start_at)).to eq(assigns(:end_at) - 1.hour) 
```

File: spec/controllers/worker_logs_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      context '1 week' do
+        let(:period) { '1w' }
+        it do
+          expect(assigns(:logs)).to match_array logs
+          expect(assigns(:start_at)).to eq assigns(:end_at) - 1.week
+        end
+      end
+    end
+
+    context 'with end_at' do
```
Comment (ganmacs) [inline spec/controllers/worker_logs_controller_spec.rb]: Please add the spec  representing if invalid time as `end_at` and `start_at` is given.

**Comment by ganmacs** [comment]
I'm sorry to be late for reviewing :bow:

**Comment by takonomura** [comment]
@ganmacs Thanks, I fixed reviewed changes.

File: lib/autoload/kuroko2/command/shell.rb [hunk 5]
```diff
@@ -50,13 +50,24 @@ def invoke(execution)
         command = execution.shell
         env     = execution.context.fetch('ENV', {})
 
+        worker_log = Kuroko2::WorkerLog.create(
```
Comment (eisuke) [inline lib/autoload/kuroko2/command/shell.rb]: I worry that a lot of records are made.
You should add a script that cleaning old records like https://github.com/cookpad/kuroko2/blob/b21749d9e9e65211db8656b5fd719e0ec2c601d8/bin/cleanup_old_instances.rb

Comment (takonomura) [reply lib/autoload/kuroko2/command/shell.rb]: I added it in https://github.com/cookpad/kuroko2/pull/98/commits/7cd2bbdf81e61e82c661d2526055b85f69aadbc3.

**Pull request merged**